modules/internal/ChapelLocale.chpl

```diff
@@ -72,7 +72,7 @@ module ChapelLocale {
     // now, we make the assumption that all requests for the number of
     // running tasks want the count from the locale the calling task is
     // running on, so the minimum possible value must be 1.
-    var runningTaskCounter : atomic int;
+    var runningTaskCounter : chpl__processorAtomicType(int);
```

Comment:
I think you could just use `atomic_int64` here like we do in DefaultAssociative.chpl

Also, I think this should have a comment about why using local atomics is the right choice